packages/react-app/src/lib/token.js

```diff
@@ -152,6 +151,9 @@ export const fetchTokenBalanceWithProvider = async (
   { address },
   account,
 ) => {
+  if (address === ADDRESS_ZERO) {
```

Comment:
Best if we don't have this logic inside this method. 
This is because there might cases when toToken.address is ADDRESS_ZERO. This is when the fromToken has not yet been bridged onto the other side. In that case toToken doesn't exist yet. So address is ADDRESS_ZERO.

Reply:
So in such cases we need to display the toToken balance as 0.

Reply:
```suggestion
  if (address === ADDRESS_ZERO && mode === 'NATIVE') {
```

As you mentioned we can do a mode check as well here.